src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/AspectSupport.java

```diff
@@ -100,20 +105,41 @@ public void buildJoins(Map<QName, DBQueryBuilderJoinCommand> singleJoins, List<D
      */
     @Override
     public void buildPredicateCommands(List<DBQueryBuilderPredicatePartCommand> predicatePartCommands)
-    {
-        DBQueryBuilderPredicatePartCommand command = new DBQueryBuilderPredicatePartCommand();
-        command.setJoinCommandType(DBQueryBuilderJoinCommandType.ASPECT);
-        command.setAlias(alias);
-        command.setType(DBQueryBuilderPredicatePartCommandType.ASPECT);
-        if(qnameIds.size() > 0)
-        {
-            command.setValues(qnameIds.toArray(new Long[]{}));
-        }
-        else
-        {
-            command.setValues(new Long[]{-1l});
-        }
-        predicatePartCommands.add(command);
-    }
+    {
+    	DBQueryBuilderPredicatePartCommand command;
+    	switch(joinType)
+    	{
+    	case NONE:
+    	case INNER:
+    	default:
```

Comment:
case default should be the last case statement. All other cases that appear after default wont be treated as default.
For example "case left:" wont get default applied to it.

Reply:
That is in fact not true as the tests prove.
It is a style issue .